app/src/main/java/com/github/onedirection/utils/LoadingDialog.java

```diff
@@ -0,0 +1,33 @@
+package com.github.onedirection.utils;
+
+import android.annotation.SuppressLint;
+import android.app.Activity;
+import android.app.AlertDialog;
+import android.view.LayoutInflater;
+
+import com.github.onedirection.R;
+
+public class LoadingDialog {
+    private final Activity activity;
+    private AlertDialog dialog;
+
+    public LoadingDialog(Activity hostActivity) {
+        activity = hostActivity;
+    }
+
+    @SuppressLint("InflateParams")
+    public void startLoadingAnimation() {
+
+        AlertDialog.Builder builder = new AlertDialog.Builder(activity);
+        LayoutInflater inflater = activity.getLayoutInflater();
```

Comment:
(Part of my previous review; dunno why github didn't include it)
You might use 
```suggestion
        LayoutInflater inflater = (LayoutInflater)context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
```
which would solve the aforementioned casting issue, as you did at [CustomCalendarView.java:111](https://github.com/Sosol02/sdp/blob/loadingScreen/app/src/main/java/com/github/onedirection/navigation/fragment/calendar/CustomCalendarView.java#L111).